cellprofiler/modules/exporttocellh5.py

```diff
@@ -366,35 +392,84 @@ def keep(column):
+                                          m.has_feature(xxx[0], xxx[1]) and
+                                         (xxx[1].startswith("AreaShape") or
+                                          xxx[1].startswith("Intensity") or
+                                          xxx[1].startswith("Texture")), columns)
+        
+        ### iterate over objects to export
+        for ch_idx, object_group in enumerate(self.objects_to_export):
+            objects_name = object_group.objects_name.value
+            
+            ### find features for that object
+            feature_cols_per_object = filter(lambda xxx: xxx[0] == objects_name, feature_cols)
+            
+            ### use first feature to get the object labels
+            c5_object_writer = c5_pos.add_region_object(objects_name)
+            first_column = feature_cols_per_object[0]
+            object_name, feature_name = first_column[:2]
+            values = m[object_name, feature_name]
+            object_labels = numpy.arange(len(values))+1
+            
+            c5_object_writer.write(t=0, object_labels=numpy.array(object_labels))
+            c5_object_writer.write_definition()
+            c5_object_writer.finalize()
+            
+            ### iterate over all cellular feature to get feature matrix
+                
+            n_features = len(feature_cols_per_object)
+            feature_names = []
+            feature_matrix = []
+            for column in feature_cols_per_object:
+                object_name, feature_name = column[:2]
                 values = m[object_name, feature_name]
-                for i, value in enumerate(values):
-                    object_number = i+1
-                    #
-                    # TODO: this is a per-object measurement for the
-                    #       object composed of the pixels whose value is
-                    #       "object_number" in the segmentation.
-                    #
-                    # e.g. cellh5.store(path, object_name, feature_name, 
-                    #                   object_number, value)
-                    #
-                    pass
+                
+                feature_names.append(feature_name)
+                feature_matrix.append(values[:, numpy.newaxis])
+                
+            feature_matrix = numpy.concatenate(feature_matrix, axis=1)
+            
+            c5_feature_writer = c5_pos.add_object_feature_matrix(object_name=object_name, feature_name="object_features", n_features=n_features, dtype=numpy.float32)
+            c5_feature_writer.write(feature_matrix)
+            c5_feature_writer.write_definition(feature_names)
+            c5_feature_writer.finalize()
+            
+            ### iterate over Location  to create bounding_box and center
+            c5_bbox = c5_pos.add_object_bounding_box(object_name=object_name)
+            
+            location_x = m[object_name, "Location_Center_X"][:, numpy.newaxis]
```

Comment:
After this is merged, I'll amend it to do a more accurate bounding box using the actual segmentation.